openssl/osx/build-openssl.sh

```diff
@@ -0,0 +1,295 @@
+# instead of `libc++` here.
+STDLIB=libc++
+
+CFLAGS="${CLANG_VERBOSE} ${SILENCED_WARNINGS} -DNDEBUG -g -O0 -pipe -fPIC -fcxx-exceptions"
+CXXFLAGS="${CLANG_VERBOSE} ${CFLAGS} -std=c++11 -stdlib=${STDLIB}"
+
+LDFLAGS="-stdlib=${STDLIB}"
+LIBS="-lc++ -lc++abi"
+
+echo "PREFIX ..................... ${PREFIX}"
+echo "USE_GIT_MASTER ............. ${USE_GIT_MASTER}"
+echo "OPENSSL_GIT_URL ........... ${OPENSSL_GIT_URL}"
+echo "OPENSSL_GIT_DIRNAME ....... ${OPENSSL_GIT_DIRNAME}"
+echo "OPENSSL_VERSION ........... ${OPENSSL_VERSION}"
+echo "OPENSSL_RELEASE_URL ....... ${OPENSSL_RELEASE_URL}"
+echo "OPENSSL_RELEASE_DIRNAME ... ${OPENSSL_RELEASE_DIRNAME}"
+echo "BUILD_MACOSX_X86_64 ........ ${BUILD_MACOSX_X86_64}"
+echo "OPENSSL_SRC_DIR ........... ${OPENSSL_SRC_DIR}"
+echo "DARWIN_X86 ..................... ${DARWIN_X86}"
+echo "DARWIN_ARM ..................... ${DARWIN_ARM}"
+echo "XCODEDIR ................... ${XCODEDIR}"
+echo "MACOSX_PLATFORM ............ ${MACOSX_PLATFORM}"
+echo "MACOSX_SYSROOT ............. ${MACOSX_SYSROOT}"
+echo "CC ......................... ${CC}"
+echo "CFLAGS ..................... ${CFLAGS}"
+echo "CXX ........................ ${CXX}"
+echo "CXXFLAGS ................... ${CXXFLAGS}"
+echo "LDFLAGS .................... ${LDFLAGS}"
+echo "LIBS ....................... ${LIBS}"
+
+
+conditionalPause
+
+echo "$(tput setaf 2)"
+echo "###################################################################"
+echo "# Fetch openssl"
+echo "###################################################################"
+echo "$(tput sgr0)"
+
+(
+    if [ -d ${OPENSSL_SRC_DIR} ]
+    then
+        rm -rf ${OPENSSL_SRC_DIR}
+    fi
+
+    mkdir -p ${OPENSSL_SRC_DIR}
+
+    #cd `dirname ${OPENSSL_SRC_DIR}`
+    cd ${OPENSSL_SRC_DIR}
+
+    if [ "${USE_GIT_MASTER}" == "YES" ]
+    then
+        git clone ${OPENSSL_GIT_URL}
+    else
+        if [ -d ${OPENSSL_RELEASE_DIRNAME} ]
+        then
+            rm -rf "${OPENSSL_RELEASE_DIRNAME}"
+        fi
+        curl --location ${OPENSSL_RELEASE_URL} --output ${OPENSSL_RELEASE_DIRNAME}.tar.gz
+        tar xvf ${OPENSSL_RELEASE_DIRNAME}.tar.gz
+        mv openssl-openssl-${OPENSSL_VERSION} openssl-${OPENSSL_VERSION}
+        pwd
+
+        rm ${OPENSSL_RELEASE_DIRNAME}.tar.gz
+
+        # Remove the version of  Test included with the release.
+        # We will replace it with version 1.7.0 in a later step.
+#        if [ -d "${OPENSSL_SRC_DIR}/gtest" ]
+#        then
+#            rm -r "${OPENSSL_SRC_DIR}/gtest"
+#        fi
+    fi
+)
+
+conditionalPause
+
+
+###################################################################
+# This section contains the build commands to create the native
+# openssl library for Mac OS X.  This is done first so we have
+# a copy of the each binaries.  It will be used in all of the
+# susequent universal builds.
+###################################################################
+
+
+
+if [ "${BUILD_MACOSX_ARM_64}" == "YES" ]
```

Comment:
@ns-vishalk `openssl` static libs are purely for debugging purposes only. ensure that you always link with the shared libs when you are building `click` etc which depend on openssl